opm/material/common/Valgrind.hpp

```diff
@@ -105,7 +105,8 @@ inline bool CheckDefined([[maybe_unused]] const T& value)
 template <class T>
 inline bool CheckAddressable([[maybe_unused]] const T& value)
 {
-#if !defined NDEBUG && HAVE_VALGRIND
+// if we run in debug mode AND we have valgrind AND we are NOT in a gpu function
+#if !defined NDEBUG && HAVE_VALGRIND && !OPM_IS_INSIDE_DEVICE_FUNCTION
```

Comment:
I do not understand this statement, apart from "Yeah"...?

The code looks fine to me.

Reply:
The very high level idea is that you don't want to call any Valgrind-function from within a device kernel. 

This is accomplished by wrapping said statement within a preprocessor `#if !OPM_IS_INSIDE_DEVICE_FUNCTION`, where 
`OPM_IS_INSIDE_DEVICE_FUNCTION` is true if and only if the function is being invoked from a device kernel. 

However, a GPU kernel is not allowed to call a function which is not decorated with `__device__`, and the function in question, `CheckAddressable` does not have any such decorator, hence it will never be called from a device kernel, in other words, the current preprocessor test is not doing anything before the `__device__` decorator (through the OPM wrapper macro) is added.

Reply:
Yes, that is kind of the gist of my comment. I'm a bit perplexed if this even solved the compilation in Debug problems, @multitalentloes ? Or is the function in question not called from any device kernel at the moment?

Reply:
I will look more into this early next week.